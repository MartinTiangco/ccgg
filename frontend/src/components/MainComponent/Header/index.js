import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Avatar, Box, Button, Card, Grid, Typography } from "@material-ui/core";
import RefreshIcon from "@material-ui/icons/Refresh";

const ddragonProfileIcon =
  "http://ddragon.leagueoflegends.com/cdn/11.9.1/img/profileicon/";

const useStyles = makeStyles((theme) => ({
  headerText: {
    color: theme.palette.secondary.main,
    fontWeight: "bold",
  },
  card: {
    padding: theme.spacing(2),
    color: theme.palette.text.secondary,
  },
  large: {
    width: theme.spacing(8),
    height: theme.spacing(8),
  },
  border: {
    backgroundColor: theme.palette.secondary.lessOpacity,
    borderRadius: "50%",
    width: theme.spacing(9),
    height: theme.spacing(9),
  },
}));

const Header = ({ summonerInfo }) => {
  const classes = useStyles();
  const { summonerName, profileId } = summonerInfo;

  const iconSrc = `${ddragonProfileIcon}${profileId}.png`;

  return (
    <Grid container item alignItems="center" justify="flex-start" spacing={2}>
      <Grid item>
        <Card className={classes.card} raised>
          <Grid
            container
            item
            alignItems="center"
            justify="flex-start"
            spacing={2}
          >
            <Grid item>
              <Box className={classes.border}>
                <Grid
                  container
                  justify="center"
                  alignItems="center"
                  style={{ height: "100%" }}
                >
                  <Grid item>
                    <Avatar
                      className={classes.large}
                      src={iconSrc}
                      alt="Profile Icon"
                    />
                  </Grid>
                </Grid>
              </Box>
            </Grid>
            <Grid item>
              <Typography className={classes.headerText} variant="h5">
                {summonerName}
              </Typography>
            </Grid>
            <Grid item>
              <Button variant="contained" color="primary">
                <RefreshIcon /> Update
              </Button>
            </Grid>
          </Grid>
        </Card>
      </Grid>
    </Grid>
  );
};

export default Header;
