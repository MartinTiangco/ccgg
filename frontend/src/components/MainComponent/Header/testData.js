export default {
  summonerName: "Plasmatops",
  profileId: 4904,
};
