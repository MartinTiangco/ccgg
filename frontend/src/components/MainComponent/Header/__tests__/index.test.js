import renderer from "react-test-renderer";
import React from "react";
import Header from "../index";

const summonerInfo = {
  summonerName: "Plasmatops",
  profileId: 4904,
};

it("snapshot with provided summonerName and profileId", () => {
  const tree = renderer.create(<Header summonerInfo={summonerInfo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
