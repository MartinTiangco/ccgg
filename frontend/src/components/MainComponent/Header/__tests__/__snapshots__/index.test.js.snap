// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot with provided summonerName and profileId 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-elevation8 MuiPaper-rounded"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-align-items-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiBox-root MuiBox-root-5 makeStyles-border-4"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
              style={
                Object {
                  "height": "100%",
                }
              }
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <div
                  className="MuiAvatar-root MuiAvatar-circle makeStyles-large-3"
                >
                  <img
                    alt="Profile Icon"
                    className="MuiAvatar-img"
                    src="http://ddragon.leagueoflegends.com/cdn/11.9.1/img/profileicon/4904.png"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <h5
            className="MuiTypography-root makeStyles-headerText-1 MuiTypography-h5"
          >
            Plasmatops
          </h5>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                />
              </svg>
               Update
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
